auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -999,16 +1168,21 @@ private LoadedIdentity parseIdentityContents(final XMLStreamReader streamReader,
             for (;;) {
                 if (streamReader.isEndElement()) {
                     if (attributes == Attributes.EMPTY && !skipAttributes) {
-                        //Since this could be a use-case wanting to modify the attributes, make sure that we have a
-                        //modifiable version of Attributes;
+                        // Since this could be a use-case wanting to modify the attributes, make sure that we have a
+                        // modifiable version of Attributes;
                         attributes = new MapAttributes();
                     }
                     return new LoadedIdentity(name, credentials, attributes, hashEncoding);
                 }
-                if (! version.getNamespace().equals(streamReader.getNamespaceURI())) {
+                if (!(version.getNamespace().equals(streamReader.getNamespaceURI())) && !(XMLSignature.XMLNS.equals(streamReader.getNamespaceURI()))) {
                     // Mixed versions unsupported.
                     throw ElytronMessages.log.fileSystemRealmInvalidContent(path, streamReader.getLocation().getLineNumber(), name);
                 }
+
+                if (version.isAtLeast(Version.VERSION_1_2) && "principal".equals(streamReader.getLocalName())) {
+                    consumeContent(streamReader);
```

Comment:
I think we need to be a bit careful here. We need to make sure that if the principal element is encountered and the version isn't at least 1.2, then the invalid content (i.e., `fileSystemRealmInvalidContent`) should be thrown.

Reply:
Added error if version mismatch

Reply:
Thanks! Just to simplify the logic and make it a bit easier to understand what's happening, I think you could update it to:

```
if ("principal".equals(streamReader.getLocalName()) {
    if (version.isAtLeast(...)) {
       ...
    } else {
       ...
}
```